test/Solution/DirectorySolutionTest.php

```diff
@@ -4,194 +4,154 @@
 
 use InvalidArgumentException;
 use PhpSchool\PhpWorkshop\Solution\DirectorySolution;
+use PhpSchool\PhpWorkshop\Utils\Path;
 use PHPUnit\Framework\TestCase;
+use Symfony\Component\Filesystem\Filesystem;
 
 class DirectorySolutionTest extends TestCase
 {
+    /**
+     * @var string
+     */
+    private $tempPath;
+
+    public function setUp(): void
+    {
+        $this->tempPath = sprintf('%s/%s', realpath(sys_get_temp_dir()), $this->getName());
+        @mkdir($this->tempPath);
+    }
+
+    public function tearDown(): void
+    {
+        $fileSystem = new Filesystem();
+        $fileSystem->remove(Path::join(realpath(sys_get_temp_dir()), 'php-school'));
```

Comment:
Probably could have used `\PhpSchool\PhpWorkshop\Utils\System::tempDir('php-school')` and in a few other places not that it really matters